### PR TITLE
Support an array of zookeepers hosts entries just like both the mesos…

### PR DESCRIPTION
**Pull request by salimane**
… and zookeeper modules

Signed-off-by: Salimane Adjao Moustapha <me@salimane.com>

### CODE REVIEW

**Comment by salimane** [comment]
*/+
wq3.00127414

File: Gemfile [hunk 1]
```diff
@@ -1,18 +1,18 @@
 source "https://rubygems.org"
 
 group :test do
-  gem "rake"
-  gem "puppet", ENV['PUPPET_VERSION'] || '~> 3.7.0'
+  gem "rake", '~> 11.2.2'
+  gem "puppet", ENV['PUPPET_VERSION'] || '~> 4.5.3'
```
Comment (perbergland-meltwater) [inline Gemfile]: if puppet is locked at 3.3.0 through the travis file, shouldn't it be that here as well?

File: Gemfile.lock [hunk 2]
```diff
@@ -51,100 +35,106 @@ GEM
...
-      coderay (~> 1.0)
-      method_source (~> 0.8)
+    pry (0.10.4)
+      coderay (~> 1.1.0)
+      method_source (~> 0.8.1)
       slop (~> 3.4)
-    puppet (3.7.4)
-      facter (> 1.6, < 3)
-      hiera (~> 1.0)
+    puppet (4.5.3)
```
Comment (perbergland-meltwater) [inline Gemfile.lock]: we run puppet 3 in puppet-site - must not lock to puppet 4 here

File: spec/classes/init_spec.rb [hunk 3]
```diff
@@ -118,6 +118,26 @@
         :ensure => 'file',
         :path   => '/lib/systemd/system/marathon.service'
         }) }
+
+      context 'master and zk servers with defaults params' do
```
Comment (perbergland-meltwater) [inline spec/classes/init_spec.rb]: unit tests = thumbs up!

File: Gemfile (hunk 1 above)
Comment (salimane) [inline Gemfile]: we have matrix of various puppet versions in the travis file. So this has been taken care of.

File: Gemfile.lock (hunk 2 above)
Comment (salimane) [inline Gemfile.lock]: in travisci , we actually delete the .lock file and run bundle install again depending on the version of puppet currently set. So this is again tested. But I will add the puppet version 3.7 in the travis build matrix

**Pull request merged**